AngularJS/src/org/angular2/inspections/AngularDuplicateAttributeInspection.java

```diff
@@ -0,0 +1,61 @@
+// Copyright 2000-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+package org.angular2.inspections;
+
+import java.util.Collection;
+import java.util.Map.Entry;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.angular2.codeInsight.attributes.Angular2AttributeDescriptor;
+import org.angular2.entities.Angular2DirectiveSelectorPsiElement;
+import org.angular2.lang.Angular2Bundle;
+import org.angular2.lang.html.parser.Angular2AttributeType;
+import org.jetbrains.annotations.NotNull;
+
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.psi.xml.XmlAttribute;
+import com.intellij.psi.xml.XmlTag;
+
+import one.util.streamex.StreamEx;
+
+public class AngularDuplicateAttributeInspection extends AngularHtmlLikeTemplateLocalInspectionTool {
+	@Override
+	protected void visitXmlTag(
+			@NotNull final ProblemsHolder problemsHolder,
+			@NotNull final XmlTag tag) {
+		StreamEx.of(tag.getAttributes())
+		        .mapToEntry(XmlAttribute::getDescriptor, Function.identity())
+		        .selectKeys(Angular2AttributeDescriptor.class)
+		        .filterKeys(AngularDuplicateAttributeInspection::shouldConsider)
+		        .collect(Collectors.groupingBy(entry -> entry.getKey().getInfo().name))
+		        .values()
+		        .stream()
+		        .filter(entries -> entries.size() > 1)
+		        .forEach(entries -> registerProblem(problemsHolder, entries));
+	}
+
+	private static boolean shouldConsider(@NotNull final Angular2AttributeDescriptor descriptor) {
+		final Angular2AttributeType type = descriptor.getInfo().type;
+		return type == Angular2AttributeType.PROPERTY_BINDING
+				|| type == Angular2AttributeType.BANANA_BOX_BINDING
+				|| type == Angular2AttributeType.REGULAR;
+	}
+
+	private static void registerProblem(
+			@NotNull final ProblemsHolder problemsHolder,
+			@NotNull final Collection<Entry<Angular2AttributeDescriptor, XmlAttribute>> entries) {
+		for (final Entry<Angular2AttributeDescriptor, XmlAttribute> entry : entries) {
+			final Angular2AttributeDescriptor descriptor = entry.getKey();
+			final XmlAttribute attribute = entry.getValue();
+			final String messageKey =
+					descriptor.getDeclaration() instanceof Angular2DirectiveSelectorPsiElement
```

Comment:
Descriptor can have multiple declarations.

Reply:
@piotrtomiak Thanks!
Wouldn't all of them be of type `Angular2DirectiveSelectorPsiElement` anyway? Or am I missing something?

Reply:
@lppedd No, the attribute declaration can come from a selector, input/output or regular HTML element.

Reply:
@piotrtomiak ahhh yes! That's right.